homeassistant/components/caldav/calendar.py

```diff
@@ -203,6 +209,35 @@ async def async_get_events(
         """Get all events in a specific time frame."""
         return await self.coordinator.async_get_events(hass, start_date, end_date)
 
+    async def async_create_event(self, **kwargs: Any) -> None:
+        """Create a new event in the calendar."""
+        _LOGGER.debug("Event: %s", kwargs)
+
+        event = Event()
+        event.add("summary", kwargs.get("summary"))
+        event.add("dtstart", kwargs.get("dtstart"))
+        event.add("dtend", kwargs.get("dtend"))
+        event.add("dtstamp", datetime.now(kwargs.get("tzinfo")))
+        event.add("description", kwargs.get("description"))
+        event.add("uid", str(uuid.uuid4()))
+
+        _rrule = kwargs.get("rrule")
+        if _rrule is not None and _rrule != "":
+            # If rrule is set as "" or None, icalendar errors out.
+            event.add("rrule", _rrule)
+
+        cal = Calendar()
+        cal.add("prodid", "-//homeassistant.io//CALDAV//EN")
+        cal.add("version", "2.0")
+        cal.add_component(event)
+        ics_data = cal.to_ical().decode("utf-8")
+
+        _LOGGER.debug("ICS data %s", ics_data)
+
+        await self.hass.async_add_executor_job(
+            self.coordinator.calendar.add_event, ics_data
```

Comment:
It would be simpler and less code to use the method that accepts keyword args for the fields of an event rather than converting fro those fields into ICS content then back. This also seems to be preferred in the examples and tutorials for `caldav` such as:
https://github.com/python-caldav/caldav/blob/a584b5e93b175f01e93005668dd047a649d484c5/examples/basic_usage_examples.py#L128

Also `save_*`  methods seem to be preferred reading the comments (though they haven't fully deprecated `add_*` to be fair)

See how the existing todo method works in `async_create_todo_item`

Reply:
I see `add_event` is an alias so you're right, thats fine. You're only responding to the last half of my comment however. That was just a smaller detail, the bigger point is the first sentence.